app/main.py

```diff
@@ -48,7 +48,10 @@ def extract_text(url: HttpUrl = Query(..., description="The URL to extract text
     # Extract images using the utility method
     images = extract_images(soup)
 
+    # Initialize OpenAI client
+    client = OpenAI(api_key=openai_api_key)
```

Comment:
Why not initialize the client inside the module that uses it?

Reply:
Initializing the OpenAI client inside the `generate_headline` function in the `app/openai_utils.py` module ensures that the client is encapsulated within the module that directly uses it. This approach promotes better separation of concerns, making the code more modular and easier to maintain. It also simplifies the function signature of `generate_headline`, as it no longer requires the client to be passed as a parameter, reducing the complexity of the function calls throughout the codebase.